client/views/hooks/useUpgradeTabParams.ts

```diff
@@ -0,0 +1,25 @@
+import { useCallback } from 'react';
+import { useQuery, UseQueryResult } from 'react-query';
+
+import { UpgradeTabVariants } from '../../../lib/getUpgradeTabType';
+import { useEndpoint } from '../../contexts/ServerContext';
+
+const placeholderData = {
+	tabType: false as false,
+	trialEndDate: undefined,
+};
+
+export const useUpgradeTabParams = (): UseQueryResult<
+	{ tabType: UpgradeTabVariants | false; trialEndDate: string | undefined },
+	unknown
```

Comment:
It's because `result` is not typed, therefore the generic parameters of the `useQuery()` call end up assigning `unknown` for `TError`. Typing like `const result: UseQueryResult<{...}, Error> = useQuery(...);` or returning it directly (without assigning to a variable) is enough.